api/data_explorer/controllers/export_url_controller.py

```diff
@@ -259,15 +302,41 @@ def _get_filter_query(filters):
     return query
 
 
+def _get_doc_generator(filter_arr):
+    if len(filter_arr) == 0:
+        return
+
+    es = Elasticsearch(current_app.config['ELASTICSEARCH_URL'])
+    es_facets = OrderedDict(current_app.config['ELASTICSEARCH_FACETS'].items())
+    filters = elasticsearch_util.deserialize(filter_arr, es_facets)
+    search_dict = DatasetFacetedSearch(filters,
+                                       es_facets).build_search().to_dict().get(
+                                           'post_filter', {})
+    search = Search(using=es, index=current_app.config['INDEX_NAME'])
+    search.update_from_dict({'post_filter': search_dict})
+    for result in search.scan():
+        yield result.to_dict()
+
+
 def export_url_post():  # noqa: E501
     _check_preconditions()
     data = json.loads(request.data)
+    filter_arr = data['filter']
+
     current_app.logger.info('Export URL request data %s' % request.data)
-    query = _get_filter_query(data['filter'])
-    cohortname = data['cohortName']
-    cohortname = cohortname.replace(" ", "_")
-    entities = _get_entities_dict(cohortname, query)
-    current_app.logger.info('Entity JSON: %s' % json.dumps(entities))
+
+    # TODO(malathir): Add extraFacets to this endpoint
+    query = _get_filter_query(filter_arr)
+    cohort_name = data['cohortName'].replace(" ", "_")
+
+    # If we have not selected a cohort there's no need accumulate
```

Comment:
```
# If a cohort was selected, get Elasticsearch documents for the cohort so we can create sample_set entity.
```

Reply:
Call is gone since I moved the call to get_doc_generator into _get_entities_dict, but added similar comment up there.